actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -122,6 +122,7 @@ def find_routes req
 
         def match_head_routes(routes, req)
           head_routes = match_routes(routes, req)
+          head_routes.delete_if { |route| route.verb == // }
```

Comment:
@pixeltrix @tsun1215 Ok :smile_cat: 

Could we perhaps delete the routes before matching them?

```ruby
head_routes.delete_if { |route| route.verb == // }
head_routes = match_routes(routes, req)
```
Since we're going to reject them, we can do it earlier to avoid matching them.